test_conformance/extensions/cl_khr_command_buffer/cl_khr_command_buffer_mutable_dispatch/mutable_command_arguments.cpp

```diff
@@ -59,48 +76,76 @@ struct MutableDispatchGlobalArguments : public BasicMutableCommandBufferTest
                 dst[tid] = src[tid];
             })";
 
-        error = create_single_kernel_helper(context, &program, &kernel, 1,
-                                            &sample_const_arg_kernel,
-                                            "sample_test");
+        cl_int error = create_single_kernel_helper(context, &program, &kernel,
+                                                   1, &sample_const_arg_kernel,
+                                                   "sample_test");
         test_error(error, "Creating kernel failed");
+        return CL_SUCCESS;
+    }
 
+    cl_int SetUpKernelArgs() override
+    {
+        cl_int error = CL_SUCCESS;
         // Create and initialize buffers
-
         MTdataHolder d(gRandomSeed);
 
-        std::vector<cl_int> srcData(num_elements);
+        srcData.resize(num_elements);
         for (size_t i = 0; i < num_elements; i++)
             srcData[i] = (cl_int)genrand_int32(d);
 
-        clMemWrapper srcBuf = clCreateBuffer(context, CL_MEM_COPY_HOST_PTR,
-                                             num_elements * sizeof(cl_int),
-                                             srcData.data(), &error);
+        in_mem = clCreateBuffer(context, CL_MEM_COPY_HOST_PTR,
+                                num_elements * sizeof(cl_int), srcData.data(),
+                                &error);
         test_error(error, "Creating src buffer");
 
-        clMemWrapper dstBuf0 =
-            clCreateBuffer(context, CL_MEM_READ_WRITE,
-                           num_elements * sizeof(cl_int), NULL, &error);
+        dstBuf0 = clCreateBuffer(context, CL_MEM_READ_WRITE,
+                                 num_elements * sizeof(cl_int), NULL, &error);
         test_error(error, "Creating initial dst buffer failed");
 
-        clMemWrapper dstBuf1 =
-            clCreateBuffer(context, CL_MEM_READ_WRITE,
-                           num_elements * sizeof(cl_int), NULL, &error);
+        dstBuf1 = clCreateBuffer(context, CL_MEM_READ_WRITE,
+                                 num_elements * sizeof(cl_int), NULL, &error);
         test_error(error, "Creating updated dst buffer failed");
 
         // Build and execute the command buffer for the initial execution
 
-        error = clSetKernelArg(kernel, 0, sizeof(srcBuf), &srcBuf);
+        error = clSetKernelArg(kernel, 0, sizeof(in_mem), &in_mem);
         test_error(error, "Unable to set src kernel arguments");
 
         error = clSetKernelArg(kernel, 1, sizeof(dstBuf0), &dstBuf0);
         test_error(error, "Unable to set initial dst kernel argument");
+        return CL_SUCCESS;
+    }
+
+    // verify the result
+    bool verify_result(const cl_mem &buffer)
+    {
+        cl_int error = CL_SUCCESS;
+        std::vector<cl_int> data(num_elements);
+        error = clEnqueueReadBuffer(queue, buffer, CL_TRUE, 0, data_size(),
```

Comment:
Prefer to define `error` the first time it's used (and in other places touched by this PR)
```suggestion
        std::vector<cl_int> data(num_elements);
        cl_int error = clEnqueueReadBuffer(queue, buffer, CL_TRUE, 0, data_size(),
```

Reply:
Done